### ISSUE TITLE
Panning/Zooming causes graph to shrink beyond it's bounds

### ISSUE THREAD

**Issue by abeatte** [jjoe64/GraphView#351]
Have a large data set.
Pinch to zoom out.
Notice the data points shrink to not fit the viewPort. 

This is using `    compile 'com.jjoe64:graphview:4.0.1'`

![screenshot 2015-06-10 20 29 58](https://cloud.githubusercontent.com/assets/2646229/8099302/b09b900a-0faf-11e5-8c71-29999518f9a8.png)

**Comment by ParasDPain** [comment]
If you set the Maximum value of the X axis using `SetMaxX()` of the data stream to the last value of the graph, it will automatically adjust the graph to fit the new `datapoint`. I suggest updating the `SetMaxX()` after each new datapoint.

**Comment by abeatte** [comment]
Excellent thought @ParasDPain but I believe hat is a method on the ViewPort. setMaxX() is used to tell how far out the viewport is. For instance, setting minX to 5 and maxX to 10 you are telling the viewport to show only data points from 5-10. What I am trying to fix is when you pinch-to-zoom/contract I need to tell the ViewPort what its maxX bounds are.

**Comment by ParasDPain** [comment]
Oh! If you want to update `MinX` and `MaxX` as the user zoom/contracts the `Viewport`, than either you're going to have to modify the `Viewport` class for this feature. A subclass won't simply work because after version 4.0 `Viewport` is only accessible via the `GraphView` class. It's all connected. Or you can benefit from an event listener for these events and update the `Viewport` as the per the touch args.

I think I may have found something you can use. The `ScaleGestureDetector` class has listeners for zoom/contract events and even provides a helper class `SimpleOnScaleGestureListener`. You can overlay an instance of detector over the viewport region and that should do it, but ofcoure you would need to override the scale feature of GraphView. I suggest setting `SetManualXBounds` and `SetManualYBounds` to `true` and turning autoscale false should do the trick. Don't think you'd be able to use a `ScrollView` at that point.
Here's some sample code taken from Android-Dev blog

```java
private class ScaleListener extends ScaleGestureDetector.SimpleOnScaleGestureListener {
    @Override
    public boolean onScale(ScaleGestureDetector detector) {
        mScaleFactor *= detector.getScaleFactor();
        // Don't let the object get too small or too large.
        mScaleFactor = Math.max(0.1f, Math.min(mScaleFactor, 5.0f));
        invalidate();
        return true;
    }
}
```

Here's the [article](http://android-developers.blogspot.co.nz/2010/06/making-sense-of-multitouch.html) (mentions `ScaleGestureDetector` at the end), and here's the [class overview](http://developer.android.com/reference/android/view/ScaleGestureDetector.html) if you'd like to directly implement this.

**Comment by jjoe64** [comment]
#328

**Comment by abeatte** [comment]
@jjoe64 #322 was also closed fixed referencing #328 but it still repros. Can we reopen this, please?
